### ISSUE TITLE
After upgrading flutter to latest stable build. It shows error while building for apk. (*.addressOf). 

### ISSUE THREAD

**Issue by itsbhanupratap** [flutter/flutter#83015]
## Steps to Reproduce
- `flutter build apk --release`

**Actual results:** 
```
../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider_windows-0.0.4+3/lib/src/path_provider_windows_real.dart:126:25: Error: The getter 'addressOf' isn't defined for the class 'GUID'.
 - 'GUID' is from 'package:win32/src/structs.dart' ('../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'addressOf'.
          knownFolderID.addressOf, KF_FLAG_DEFAULT, NULL, pathPtrPtr);  
                        ^^^^^^^^^                                       
../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:1067:7: Error: The getter 'addressOf' isn't defined for the class 'LOGFONT'.
 - 'LOGFONT' is from 'package:win32/src/structs.dart' ('../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'addressOf'.
      addressOf.cast<Uint8>().elementAt(28).cast<Utf16>(); 


../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:1111:7: Error: Struct 'ENUMLOGFONTEX' is empty. Empty structs are undefined behavior.
class ENUMLOGFONTEX extends Struct {                                    
      ^                                                                 
../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2835:7: Error: Struct 'BLUETOOTH_PIN_INFO' is empty. Empty structs are undefined behavior.
class BLUETOOTH_PIN_INFO extends Struct {                               
      ^                                    

../../../../snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/allocation.dart:47:33: Error: Expected type 'T' to be a valid and instantiated subtype of 'NativeType'.
  final int totalSize = count * sizeOf<T>();                            
                                ^              
```

`flutter doctor -v` 

```
Flutter 2.2.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b22742018b (5 days ago) • 2021-05-14 19:12:57 -0700
Engine • revision a9d88a4d18
Tools • Dart 2.13.0

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.0, on Linux, locale en_GB.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] Android Studio (version 4.1)
[✓] Connected device (1 available)

• No issues found!
```

</details>

**Comment by dcharkes** [comment]
Did a number of breaking changes: https://github.com/dart-lang/sdk/issues/44621, https://github.com/dart-lang/sdk/issues/44622, and https://github.com/dart-lang/sdk/issues/40667.

Your dependencies `package:ffi` and `package:win32` have newer versions which use new `dart:ffi` features. Please update your dependencies and you should be good to go.

**Comment by itsbhanupratap** [comment]
Is there any way to check which plugin is using that dependencies to upgrade that specific plugin.
I don't have ffi and win32 as a dependency.

My pubspec.yaml
```
  flutter_screenutil: ^4.0.2+8
  get_it: ^6.1.1
  cupertino_icons: ^1.0.0
  stacked: ^2.1.6
  auto_route: ^1.0.2
  stacked_services: ^0.8.6
  injectable:
  pin_code_fields: ^7.1.0
  date_time_picker: ^1.1.1
  http: ^0.12.1
  lottie: ^0.7.0+1
  cached_network_image: ^2.5.1
  sms_autofill: ^2.0.0
  simple_speed_dial: ^0.1.2
  flutter_datetime_picker: ^1.5.1
  syncfusion_flutter_datepicker: ^19.1.59-beta
  shimmer: ^1.1.2
  file_picker: ^2.1.7
  firebase_core: ^0.5.3
  firebase_storage: ^5.0.0
  firebase_messaging: ^7.0.3
  url_launcher: ^5.7.10
  video_player: ^2.1.1
  share: ^0.6.5+4
  awesome_dialog: ^2.0.0
  flutter_easyloading: ^3.0.0
  slugify: ^2.0.0
  multi_select_flutter: ^4.0.0
  modal_bottom_sheet: ^2.0.0
  badges: ^2.0.1
  introduction_screen: ^2.1.0
  flutter_hooks: ^0.17.0
  observable_ish: ^3.0.0
  pub_semver: ^1.4.4
  package_info: ^0.4.3+2
  flutter_overlay_loader: ^0.1.1
  loading_indicator: ^2.0.1
  android_play_install_referrer: ^0.1.1
  material_floating_search_bar: ^0.3.4
  stream_chat_flutter: ^1.5.2
  stream_chat_persistence: ^1.5.1
  flutter_local_notifications: ^4.0.1+2 
  flutter_svg: ^0.19.1

dev_dependencies:
  flutter_test:
    sdk: flutter
  lint: ^1.0.0
  build_runner:
  auto_route_generator:
  injectable_generator:

```

**Comment by dcharkes** [comment]
`dart pub deps` or `flutter pub deps`.

Try `flutter pub upgrade` for upgrading everything that is not constrained.

**Comment by stuartmorgan** [comment]
Whatever is pulling an old version of path_provider will need to be updated; that version doesn't support 2.2 (because of the changes referenced above).

**Comment by stuartmorgan** [comment]
Actually, I forgot we published `path_provider_windows` 0.0.5`. Just updating to that, which won't require updating any of your dependencies, should fix this.
